src/main/java/org/cloudbees/literate/jenkins/promotions/PromotionProject.java

```diff
@@ -489,7 +490,11 @@ public String getId() {
     }
 
     public DescriptorImpl getDescriptor() {
-        return (DescriptorImpl) Jenkins.getInstance().getDescriptorOrDie(getClass());
+        Jenkins j = Jenkins.getInstance();
+        if (j == null) {
+            throw new IllegalStateException(); // TODO 1.590+ getActiveInstance
```

Comment:
please add a message - if a user ever sees this having ("Jenkins is starting up or shutting down") is highly valuable.